eva_cttv_pipeline/evidence_string_generation/pipeline.nf

```diff
@@ -0,0 +1,266 @@
+   script:
+   """
+   \${PYTHON_BIN} \${CODE_ROOT}/consequence_prediction/run_repeat_expansion_variants.py \
+        --clinvar-xml ${clinvarXml} \
+        --output-consequences consequences_repeat.tsv
+
+    # create an empty file if nothing generated
+    [[ -f consequences_repeat.tsv ]] || touch consequences_repeat.tsv
+   """
+}
+
+/*
+ * Run consequence and gene mapping for structural variants (i.e. no complete coordinates and not
+ * known repeat expansions).
+ */
+process runStructural {
+   clusterOptions "-o ${batchRoot}/logs/consequence_structural.out \
+                   -e ${batchRoot}/logs/consequence_structural.err"
+
+   publishDir "${batchRoot}/gene_mapping",
+       overwrite: true,
+       mode: "copy",
+       pattern: "*.tsv"
+
+   input:
+   path clinvarXml
+
+   output:
+   path "consequences_structural.tsv", emit: consequencesStructural
+
+   script:
+   """
+   \${PYTHON_BIN} \${CODE_ROOT}/consequence_prediction/run_structural_variants.py \
+        --clinvar-xml ${clinvarXml} \
+        --output-consequences consequences_structural.tsv
+
+    # create an empty file if nothing generated
+    [[ -f consequences_structural.tsv ]] || touch consequences_structural.tsv
+   """
+}
+
+/*
+ * Unite results of consequence mapping.
+ */
+process combineConsequences {
+    input:
+    path consequencesSnp
+    path consequencesRepeat
+    path consequencesStructural
+
+    output:
+    path "consequences_combined.tsv", emit: consequencesCombined
+
+    script:
+    """
+    cat ${consequencesRepeat} ${consequencesSnp} ${consequencesStructural} > consequences_combined.tsv
+    """
+}
+
+/*
+ * Generate the evidence strings for submission to Open Targets.
+ */
+process generateEvidence {
+    clusterOptions "-o ${batchRoot}/logs/evidence_string_generation.out \
+                    -e ${batchRoot}/logs/evidence_string_generation.err"
+
+    publishDir "${batchRoot}/evidence_strings",
+        overwrite: true,
+        mode: "copy",
+        pattern: "*.json"
+
+    input:
+    path clinvarXml
+    path jsonSchema
+    path consequenceMappings
+
+    output:
+    path "evidence_strings.json", emit: evidenceStrings
+
+    script:
+    """
+    \${PYTHON_BIN} \${CODE_ROOT}/bin/evidence_string_generation.py \
+        --clinvar-xml ${clinvarXml} \
+        --efo-mapping \${BATCH_ROOT_BASE}/manual_curation/latest_mappings.tsv \
+        --gene-mapping ${consequenceMappings} \
+        --ot-schema ${jsonSchema} \
+        --out .
+    """
+}
+
+/*
+ * Check that the generated evidence strings do not contain any duplicated evidence strings.
+ */
+process checkDuplicates {
+    input:
+    path evidenceStrings
+
+    script:
+    """
+    (jq --arg sep \$'\t' -jr \
+        '.datatypeId,\$sep,.studyId,\$sep,.targetFromSourceId,\$sep,.variantId,\$sep,.variantFunctionalConsequenceId,\$sep,.diseaseFromSourceMappedId,\sep,.diseaseFromSource,"\n"' \
+        ${evidenceStrings} | sort | uniq -d > duplicates.tsv) \
+    || [[ -z duplicates.tsv ]]
```

Comment:
I don't think we need the `||` since nextflow should fail if the first command fail.